catkit/hardware/mcpherson/scan_controller_789a4.py

```diff
@@ -0,0 +1,616 @@
+import enum
+import functools
+import re
+import time
+import warnings
+
+from catkit.interfaces.Instrument import Instrument
+import numpy as np
+import pyvisa
+
+
+DEFAULT_POLL = 0.8  # As documented in manual (seconds).
+DEFAULT_POLL_TIMEOUT = 60  # This is not the comms timeout but that allowed for total polling duration (seconds).
+
+
+class ASCIIControlCodes(enum.Enum):
+    ENTER_CMD_MODE = " "  # AKA INIT.
+    # INIT = ' '
+    EXIT_CMD_MODE = chr(3)  # AKA RESET.
+    # RESET = chr(3)  # Stops motion, sets counter to 0 (so man says, but doesn't), assumes idle state.
+    SOFT_STOP = '@'  # Causes deacceleration to stop.
+    DISABLE_HOME_SWITCH = "A0"
+    ENABLE_HOME_SWITCH = "A8"
+    ENABLE_HIGH_ACCURACY_HOMING = "A24"
+    CLEAR = "C1"
+    FIND_EDGE = 'F'
+    # EXEC_INTERNAL_FUNC = 'G'
+    START_VELOCITY = 'I'
+    RAMP_SLOPE = 'K'
+    PROGRAM_MODE = 'P'
+    CONSTANT_VELOCITY_MOVE = 'M'
+    SAVE = 'S'
+    SCAN_SPEED = 'V'
+    PAUSE = 'W'  # Wait/sleep.
+    READ_PARAMS = 'X'
+    GET_LIMIT_SWITCH_STATUS = ']'  # 0=no lim encountered, 32=home lim enc., 64=high lim enc., 128=low lim enc.
+    SCAN_UP = '+'
+    SCAN_DOWN = '-'
+    GET_MOTION_STATUS = '^'  # 0=not moving, 1=moving, 2=high in const vel, 16=slewing ramping complete.
+
+
+class LimitSwitchStatus(enum.Enum):
+    # NOTE: It's possible to be both HOME & LOW as the HOME region is wide enough to include the LOW limit.
+    NO_LIMIT = 0  # Not at any of the below. NOTE: Could be HOME but requires home switch to be enabled.
```

Comment:
I don't like this naming, but couldn't think of anything better at the time, though perhaps ``INBETWEEN`` is better? 🤷